packages/data-broker/src/services/weather/WeatherService.ts

```diff
@@ -36,20 +34,32 @@ export class WeatherService extends Service {
     this.dateSanitiser = new DateSanitiser();
   }
 
+  public pullAnalytics(dataElements: DataElement[], options: PullOptions) {
+    return this.pull(dataElements, 'dataElement', options);
+  }
+
+  public pullEvents(dataGroups: DataGroup[], options: PullOptions) {
+    return this.pull(dataGroups, 'dataGroup', options);
+  }
+
+  public async pullSyncGroupResults(): Promise<never> {
+    throw new Error('pullSyncGroupResults is not supported in WeatherService');
+  }
+
   /**
    * Note, if no period specified will return current weather
    */
-  public async pull(
+  private async pull(
```

Comment:
This is the only case where I left the original, data-source-level `pull()` methods in place. Specific logic is repeated across data sources so it kind of makes sense that way.